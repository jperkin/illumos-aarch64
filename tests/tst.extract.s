/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2017 Joyent, Inc.
 */

/*
 * C3.4.3 Extract
 */
.text
.globl libdis_test
.type libdis_test, @function
libdis_test:
	/*
	 * C5.6.67 EXTR
	 * C5.6.152 ROR (immediate) (alias)
	 */
	extr	w0, w15, wzr, #0
	extr	w0, w15, wzr, #31
	extr	x0, x15, xzr, #0
	extr	x0, x15, xzr, #63
	/* Test for ROR (Rn == Rm) */
	extr	w0, wzr, wzr, #0
	extr	w0, wzr, wzr, #31
	extr	x0, xzr, xzr, #0
	extr	x0, xzr, xzr, #31
.size libdis_test, [.-libdis_test]
