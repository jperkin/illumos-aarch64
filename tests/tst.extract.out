    libdis_test:      e0 01 9f 13  extr w0, w15, wzr, #0
    libdis_test+0x4:  e0 7d 9f 13  extr w0, w15, wzr, #31
    libdis_test+0x8:  e0 01 df 93  extr x0, x15, xzr, #0
    libdis_test+0xc:  e0 fd df 93  extr x0, x15, xzr, #63
    libdis_test+0x10: e0 03 9f 13  ror w0, wzr, #0
    libdis_test+0x14: e0 7f 9f 13  ror w0, wzr, #31
    libdis_test+0x18: e0 03 df 93  ror x0, xzr, #0
    libdis_test+0x1c: e0 7f df 93  ror x0, xzr, #31
